src/stores/documents.ts

```diff
@@ -0,0 +1,84 @@
+  const baseURL = config.public.docApiURL
+  const docApiKey = config.public.docApiKey
+
+  /** Array of Document objects returned from the DRS API */
+  const documents = ref([] as Array<DocumentIF>)
+
+  /** Returns True if the user is a staff account and the document records feature flag is enabled */
+  const enableDocumentRecords = computed(() => useBcrosAccount().isStaffAccount &&
+    !!useBcrosLaunchdarkly().getFeatureFlag('enable-document-records'))
+
+  /**
+   * Sends a GET request to fetch a document from the specified API endpoint (Document Record Service).
+   *
+   * @param params - The parameters for the document request, including document class, type, and consumer information.
+   * @returns A promise that resolves to either an ApiResponseIF on success or an ApiErrorIF on failure.
+  */
+  const getCorpDocuments = async (params: DocumentRequestIF): Promise<ApiResponseOrError> => {
+    const options = {
+      method: 'GET',
+      headers: { 'x-apikey': `${docApiKey}` }
+    }
+
+    const {
+      pageNumber,
+      consumerDocumentId,
+      consumerFilename,
+      documentType,
+      consumerIdentifier,
+      queryStartDate,
+      queryEndDate
+    } = params
+
+    // Construct query parameters
+    const queryParams = new URLSearchParams()
+    if (consumerDocumentId) { queryParams.append('consumerDocumentId', consumerDocumentId) }
+    if (consumerIdentifier) { queryParams.append('consumerIdentifier', consumerIdentifier) }
+    if (consumerFilename) { queryParams.append('consumerFilename', consumerFilename) }
+    if (documentType) { queryParams.append('documentType', documentType) }
+    if (queryStartDate) { queryParams.append('queryStartDate', queryStartDate) }
+    if (queryEndDate) { queryParams.append('queryEndDate', queryEndDate) }
+    if (pageNumber) { queryParams.append('pageNumber', pageNumber.toString()) }
+
+    // Build the full URL
+    const url = `${baseURL}/searches/CORP?${queryParams.toString()}`
+
+    try {
+      const { data } = await useBcrosFetch<ApiResponseIF>(url, options)
+      documents.value = data.value as unknown as DocumentIF[]
+    } catch (error) {
+      const axiosError = error as AxiosError
+      return {
+        message: axiosError.message,
+        status: axiosError.response?.status,
+        statusText: axiosError.response?.statusText
+      }
+    }
+  }
+
+  /**
+   * Retrieves the consumerDocumentId of a document that matches the given filingId.
+   *
+   * @param documents - An array of Document objects to search through.
+   * @param filingId - The filingId to match against the consumerReferenceId.
+   * @returns The consumerDocumentId of the matching document, or undefined if no match is found.
+  */
+  const getDocIdByFilingId = (documents: DocumentIF[], filingId: number): string | undefined => {
+    const document = documents?.find(doc =>
+      doc.consumerReferenceId === filingId.toString())
+    return document ? document.consumerDocumentId : undefined
```

Comment:
You could use optional chaining here.

I'm wary of returning undefined, since it may be non-reactive.

Reply:
Definitely, will do!

Reply:
Updated

Reply:
If possible, please test with `document` or `document.consumerDocumentId` equal to undefined, and check if the computed in CommonTemplate.vue still works correctly.

In the past (albeit in Vue2), I've found that undefined values aren't reactive, meaning getters and watchers don't update as expected.